solidity/contracts/deposit/DepositUtils.sol

```diff
@@ -396,6 +399,34 @@ library DepositUtils {
         return _postCallBalance.sub(_preCallBalance);
     }
 
+    /// @notice     Adds a given amount to an addresses withdrawal allowance.
+    /// @dev        Withdrawals can only happen when a contract is in an end-state.
+    function enableWithdrawal(DepositUtils.Deposit storage _d, address _withdrawer, uint256 _amount) internal {
+        _d.withdrawalAllowances[_withdrawer] = _d.withdrawalAllowances[_withdrawer].add(_amount);
+    }
+
+    /// @notice     Withdraw caller's allowance.
+    /// @dev        Withdrawals can only happen when a contract is in an end-state.
+    function withdrawFunds(DepositUtils.Deposit storage _d) internal {
+        uint256 available = _d.withdrawalAllowances[msg.sender];
+
+        require(_d.inEndState(), "Contract not yet terminated");
+        require(available > 0, "Nothing to withdraw");
+        require(address(this).balance >= available, "Insufficient contract balance");
+
+        // zero-out to prevent reentrancy
+        _d.withdrawalAllowances[msg.sender] = 0;
+
+        /* solium-disable-next-line security/no-call-value */
+        msg.sender.call.value(available)("");
+    }
+
+    /// @notice     Get the caller's withdraw allowance.
+    /// @return     The caller's withdrawable allowance in wei.
```

Comment:
We're back-and-forthing between “withdraw allowance” and “withdrawable allowance”. I'm fine with “withdraw allowance”, but let's use it everywhere.